src/views/home/components/bridge-form/bridge-form.view.tsx

```diff
@@ -80,19 +103,87 @@ const BridgeForm: FC<BridgeFormProps> = ({ account, formData, resetForm, onSubmi
     }
   };
 
-  const onTokenDropdownClick = (chain: Chain) => {
-    setTokenList(getChainTokens(chain));
+  const onTokenDropdownClick = () => {
+    setTokenList(tokensWithBalance);
   };
 
-  const onTokenClick = (token: Token) => {
+  const onTokenListTokenSelected = (token: TokenWithBalance) => {
     setToken(token);
     setTokenList(undefined);
     setAmount(undefined);
   };
 
-  const onInputChange = ({ amount, error }: { amount?: BigNumber; error?: string }) => {
-    setAmount(amount);
-    setInputError(error);
+  const updateTokenList = (tokensWithBalance: TokenWithBalance[], tokenListSearchTerm: string) => {
+    const filteredTokens = tokensWithBalance.filter(getTokenFilterByTerm(tokenListSearchTerm));
+    setTokenList(filteredTokens);
+    setTokenListCustomToken({ status: "pending" });
+    if (chains && ethersUtils.isAddress(tokenListSearchTerm) && filteredTokens.length === 0) {
+      setTokenListCustomToken({ status: "loading" });
+      void getTokenFromAddress({
+        address: tokenListSearchTerm,
+        chain: chains.from,
+      })
+        .then((token) => {
+          getTokenBalance(token, chains.from)
+            .then((balance) => {
+              callIfMounted(() => {
+                const tokenWithBalance: TokenWithBalance = { ...token, balance };
+                setTokenList([tokenWithBalance]);
+                callIfMounted(() => {
+                  setTokenListCustomToken({ status: "successful", data: tokenWithBalance });
+                });
+              });
+            })
+            .catch(() => {
+              callIfMounted(() => {
+                const tokenWithBalance: TokenWithBalance = { ...token, balance: undefined };
+                setTokenList([tokenWithBalance]);
+              });
+            });
+        })
+        .catch(() =>
+          callIfMounted(() => {
+            setTokenListCustomToken({
+              status: "failed",
+              error: `The token can not be imported: A problem occurred calling the provided contract on the ${getChainName(
+                chains.from
+              )} chain with id ${chains.from.chainId}`,
+            });
+          })
+        );
+    }
+  };
+
+  const onTokenListClosed = () => {
+    setTokenList(undefined);
```

Comment:
We don't need to set to `undefined` the `tokenList` when we load a `customToken`.  In fact, this responsibility is from the `TokenList` component. You are trying to say: "Don't render the token list but display the custom token instead. If it's loading display the loading indicator". The data is still there, but the component should render the view accordingly.

Reply:
I don't understand this comment. Here we're setting the `tokenList` to undefined when the modal is closed. How is this related to the loading of a custom token?